cmd/internal/opentelemetry/wrapperMux.go

```diff
@@ -0,0 +1,203 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentelemetry
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/google/cadvisor/version"
+	"os"
+
+	"go.opentelemetry.io/otel/api/correlation"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/propagation"
+	apiTrace "go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/plugin/othttp"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"google.golang.org/grpc/credentials"
+	"k8s.io/klog/v2"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const traceServiceName = "cadvisor"
+
+var traceProbability = flag.Float64("trace_probability", 1, "used for sampler (0=never, 1=always and probability is 0.1 to 0.99)")
+
+func InitTrace(traceEndpoint, collectorCert string) *otlp.Exporter {
+	klog.Infof("write telemetry to %s", traceEndpoint)
+	exporterOptions := []otlp.ExporterOption{otlp.WithAddress(traceEndpoint)}
+
+	if collectorCert != "" {
+		creds, err := credentials.NewClientTLSFromFile(collectorCert, "")
+		if err != nil {
+			klog.Fatal(err)
+		}
+		exporterOptions = append(exporterOptions, otlp.WithTLSCredentials(creds))
+	} else {
+		exporterOptions = append(exporterOptions, otlp.WithInsecure())
+	}
+
+	exporter, err := otlp.NewExporter(exporterOptions...)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	sampler := getTraceSamplerProbability() //*traceSampler, *traceSamplerParam)
+	providerOpts := []trace.ProviderOption{
+		trace.WithBatcher(exporter),
+		trace.WithConfig(trace.Config{DefaultSampler: sampler}),
+	}
+
+	traceProvider, err := trace.NewProvider(providerOpts...)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	global.SetTraceProvider(traceProvider)
+
+	apiB3Single, apiB3, traceCtx, corrCtx := apiTrace.B3{SingleHeader: false}, apiTrace.B3{SingleHeader: false}, apiTrace.DefaultHTTPPropagator(), correlation.DefaultHTTPPropagator()
+	extractor := extractorFilter{[]propagation.HTTPExtractor{apiB3Single, apiB3, traceCtx, corrCtx}}
+	global.SetPropagators(propagation.New(propagation.WithInjectors(apiB3, traceCtx, corrCtx), propagation.WithExtractors(extractor)))
+	return exporter
+}
+
+type extractorFilter struct {
+	extractors []propagation.HTTPExtractor
+}
+
+func (e extractorFilter) Extract(orgCtx context.Context, supplier propagation.HTTPSupplier) context.Context {
+	for _, extractor := range e.extractors {
+		if newCtx := extractor.Extract(orgCtx, supplier); apiTrace.RemoteSpanContextFromContext(newCtx).IsValid() {
+			return newCtx
+		}
+	}
+	return orgCtx
+}
+
+func getTraceSamplerProbability() trace.Sampler {
+	probability := *traceProbability
+	if probability < 0 && probability > 1 {
+		klog.Errorf("invalid probability value %f, probability is allowed between 0.0 and 1.0", probability)
```

Comment:
Shouldn't you `return trace.NeverSample()` here? Otherwise we are going to hit `return trace.ProbabilitySampler(probability)` with invalid probability.